### PR TITLE
Add sms spam dataset

### PR DESCRIPTION
**Pull request by Chandu-4444**
I've added SMSSpamCollection dataset and modified the structure of the features returned by the titanic dataset (Each column is a single sample)

### CODE REVIEW

File: src/SMSSpamCollection/SMSSpamCollection.jl [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+3. Limitation of Liability. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR THEIR AFFILIATES, OR THE CORPUS CONTRIBUTING EDITORS, BE LIABLE FOR ANY INDIRECT, SPECIAL, INCIDENTAL OR CONSEQUENTIAL DAMAGES, INCLUDING, WITHOUT LIMITATION, DAMAGES FOR LOSS OF GOODWILL OR ANY AND ALL OTHER COMMERCIAL DAMAGES OR LOSSES, EVEN IF ADVISED OF THE POSSIBILITY THEREOF, AND REGARDLESS OF WHETHER ANY CLAIM IS BASED UPON ANY CONTRACT, TORT OR OTHER LEGAL OR EQUITABLE THEORY, RELATING OR ARISING FROM THE CORPUS, YOUR CORPUS USE OR THIS LICENSE AGREEMENT.
+
+# Interface
+- [`SMSSpamCollection.features`](@ref)
+- [`SMSSpamCollection.targets`](@ref)
+
+"""
+
```
Comment (CarloLucibello) [inline src/SMSSpamCollection/SMSSpamCollection.jl:85]:
```suggestion
```

File: src/SMSSpamCollection/SMSSpamCollection.jl [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+3. Limitation of Liability. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR THEIR AFFILIATES, OR THE CORPUS CONTRIBUTING EDITORS, BE LIABLE FOR ANY INDIRECT, SPECIAL, INCIDENTAL OR CONSEQUENTIAL DAMAGES, INCLUDING, WITHOUT LIMITATION, DAMAGES FOR LOSS OF GOODWILL OR ANY AND ALL OTHER COMMERCIAL DAMAGES OR LOSSES, EVEN IF ADVISED OF THE POSSIBILITY THEREOF, AND REGARDLESS OF WHETHER ANY CLAIM IS BASED UPON ANY CONTRACT, TORT OR OTHER LEGAL OR EQUITABLE THEORY, RELATING OR ARISING FROM THE CORPUS, YOUR CORPUS USE OR THIS LICENSE AGREEMENT.
+
+# Interface
```
Comment (CarloLucibello) [inline src/SMSSpamCollection/SMSSpamCollection.jl:80]:
```suggestion
# Interface

```

File: src/SMSSpamCollection/SMSSpamCollection.jl [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+julia> targets = SMSSpamCollection.targets()
+
+julia> summary(targets)
+"1×5574 Matrix{Any}"
+
+julia> targets[1]
+"ham"
+```
+"""
+function targets(; dir = nothing)
```
Comment (CarloLucibello) [inline src/SMSSpamCollection/SMSSpamCollection.jl:113]: why you provide a `dir` argument and then don't use it?

File: src/Titanic/Titanic.jl [hunk 4]
```diff
@@ -152,8 +152,13 @@ julia> summary(features)
...
-    titanic_data = readdlm(DATA, ',')
-    reshape(Matrix(hcat(titanic_data[2:end, 1], titanic_data[2:end, 3:12])), (11, 891))
+    data = readdlm(DATA, ',')
+    data = hcat(data[2:end, 1], data[2:end, 3:end])
+    titanic_data = []
+    for i in 1:891
+        append!(titanic_data, data[i,:])
+    end
+    reshape(titanic_data, (11, 891))
 end
```
Comment (CarloLucibello) [inline src/Titanic/Titanic.jl:162]: please do this in a separate PR where you specify exactly what was wrong with the previous implementation

**Comment by CarloLucibello** [comment]
Let's use DataDeps and download the csv only if needed instead of adding it to the repo

**Comment by CarloLucibello** [comment]
Also tests are missing.

File: src/Titanic/Titanic.jl [hunk 5]
```diff
@@ -152,8 +152,13 @@ julia> summary(features)
...
 function features()
-    titanic_data = readdlm(DATA, ',')
-    reshape(Matrix(hcat(titanic_data[2:end, 1], titanic_data[2:end, 3:12])), (11, 891))
+    data = readdlm(DATA, ',')
+    data = hcat(data[2:end, 1], data[2:end, 3:end])
+    titanic_data = []
+    for i in 1:891
+        append!(titanic_data, data[i,:])
+    end
+    reshape(titanic_data, (11, 891))
```
Comment (johnnychen94) [inline src/Titanic/Titanic.jl:161]:
```suggestion
    permutedims(data, (2, 1))
```

**Comment by CarloLucibello** [comment]
> Thanks! I'm sorry that I didn't check it very carefully when I merged https://github.com/JuliaML/MLDatasets.jl/pull/91

I see the problem now, not your fault, I cannot believe the author of that PR didn't even check the output

```julia
julia> Titanic.features()
11×891 Matrix{Any}:
  1  12  23  34  45  56  67  78  89  100  111  122  133  144  155  166  177  188  199  210  221  232  243  …  "S"  "S"  "C"  "S"  "S"  "S"  "C"  "S"  "S"  "S"  "S"  "S"  "C"  "S"  "Q"  "S"  "C"  "C"  "S"  "S"
  2  13  24  35  46  57  68  79  90  101  112  123  134  145  156  167  178  189  200  211  222  233  244     "S"  "S"  "S"  "S"  "C"  "Q"  "S"  "Q"  "S"  "S"  "S"  "C"  "S"  "S"  "S"  "S"  "S"  "C"  "S"  "S"
  3  14  25  36  47  58  69  80  91  102  113  124  135  146  157  168  179  190  201  212  223  234  245     "S"  "S"  "S"  "S"  "S"  "S"  "S"  "S"  "S"  "S"  "S"  "S"  "S"  "S"  "C"  "S"  "C"  "S"  "S"  "S"
  4  15  26  37  48  59  70  81  92  103  114  125  136  147  158  169  180  191  202  213  224  235  246     "S"  "C"  "S"  "S"  "Q"  "S"  "S"  "S"  "C"  "C"  "S"  "S"  "S"  "C"  "Q"  "C"  "S"  "S"  "S"  "S"
  5  16  27  38  49  60  71  82  93  104  115  126  137  148  159  170  181  192  203  214  225  236  247     "S"  "S"  "Q"  "S"  "S"  "S"  "S"  "S"  "S"  "S"  "S"  "S"  "S"  "S"  ""   "S"  "S"  "S"  "S"  "S"
  6  17  28  39  50  61  72  83  94  105  116  127  138  149  160  171  182  193  204  215  226  237  248  …  "S"  "S"  "C"  "C"  "S"  "C"  "C"  "S"  "S"  "S"  "S"  "S"  "S"  "S"  "C"  "S"  "C"  "S"  "C"  "Q"
  7  18  29  40  51  62  73  84  95  106  117  128  139  150  161  172  183  194  205  216  227  238  249     "S"  "S"  "S"  "C"  "S"  "S"  "S"  "S"  "S"  "Q"  "Q"  "C"  "S"  "S"  "S"  "C"  "S"  "S"  "C"  "S"
  8  19  30  41  52  63  74  85  96  107  118  129  140  151  162  173  184  195  206  217  228  239  250     "S"  "S"  "C"  "S"  "S"  "S"  "S"  "S"  "C"  "S"  "S"  "S"  "S"  "S"  "C"  "C"  "S"  "S"  "S"  "S"
  9  20  31  42  53  64  75  86  97  108  119  130  141  152  163  174  185  196  207  218  229  240  251     "C"  "S"  "S"  "S"  "S"  "S"  "S"  "S"  "Q"  "Q"  "C"  "S"  "S"  "S"  "S"  "S"  "S"  "C"  "S"  "S"
 10  21  32  43  54  65  76  87  98  109  120  131  142  153  164  175  186  197  208  219  230  241  252     "Q"  "C"  "C"  "S"  "S"  "S"  "S"  "S"  "Q"  "S"  "Q"  "S"  "S"  "S"  "S"  "S"  "S"  "S"  "S"  "C"
 11  22  33  44  55  66  77  88  99  110  121  132  143  154  165  176  187  198  209  220  231  242  253  …  "C"  "S"  "Q"  "S"  "S"  "S"  "S"  "S"  "S"  "C"  "S"  "S"  "S"  "S"  "C"  "S"  "S"  "S"  "C"  "Q"
```

File: src/SMSSpamCollection/SMSSpamCollection.jl (hunk 3 above)
Comment (Chandu-4444) [reply src/SMSSpamCollection/SMSSpamCollection.jl]: Um, I removed it now.

**Comment by Chandu-4444** [comment]
> Let's use DataDeps and download the csv only if needed instead of adding it to the repo

Yeah, I want to do this but I got a few errors while reading the downloaded CSV through `readdlm` so I need to tweak (Read the csv with `readlines()` and saved the output into a csv, this solved those errors) the original csv and use it. I think I can use the DataDeps and do the above before processing the csv?

**Comment by johnnychen94** [comment]
Hmmm, the doctest failure is indeed strange... I'll skip the doctest for SMDSpamCollection and merge this PR now.
